dashboard/app/models/school_info.rb

```diff
@@ -82,6 +82,13 @@ def school_state=(input)
   # Only sync from school on create to avoid unintended updates to old data
   before_validation :sync_from_schools, on: :create
 
+  before_validation :readonly?, on: :update
```

Comment:
I'm not sure this is needed. The `readonly?` method below doesn't actually throw, it overrides the default [`ActiveRecord::Core#readonly?`](https://apidock.com/rails/ActiveRecord/Core/readonly%3F) method called in the code sample you linked.

The test passes for me without this line. Additionally, `readonly?` returns true in the update case, which would be the reverse of what we want. If the `before_validation` callback returned a false value it _used_ to halt the callback chain, but [that's no longer true in Rail 5](https://blog.bigbinary.com/2016/02/13/rails-5-does-not-halt-callback-chain-when-false-is-returned.html).

Reply:
@joshlory  I thought we want SchoolInfo to be readonly in update case so the data isn't altered.

Reply:
Correct! Looks like this is fixed with commit 828a16b9fe2da71ea8c038f5ad75b7725a9023fc.